packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/helpers/V1_RawLambdaResolver.ts

```diff
@@ -0,0 +1,385 @@
+        ) ?? unitPath;
+    }
+    return spec;
+  }
+  visit_KeyExpression(spec: V1_KeyExpression): V1_ValueSpecification {
+    spec.key.accept_ValueSpecificationVisitor(this);
+    spec.expression.accept_ValueSpecificationVisitor(this);
+    return spec;
+  }
+  visit_PrimitiveType(spec: V1_PrimitiveType): V1_ValueSpecification {
+    return spec;
+  }
+  visit_TDSAggregateValue(spec: V1_TDSAggregateValue): V1_ValueSpecification {
+    spec.pmapFn.accept_ValueSpecificationVisitor(this);
+    spec.aggregateFn.accept_ValueSpecificationVisitor(this);
+    return spec;
+  }
+  visit_TDSColumnInformation(
+    spec: V1_TDSColumnInformation,
+  ): V1_ValueSpecification {
+    spec.columnFn.accept_ValueSpecificationVisitor(this);
+    return spec;
+  }
+  visit_TDSSortInformation(spec: V1_TDSSortInformation): V1_ValueSpecification {
+    return spec;
+  }
+  visit_TdsOlapRank(spec: V1_TdsOlapRank): V1_ValueSpecification {
+    spec.function.accept_ValueSpecificationVisitor(this);
+    return spec;
+  }
+  visit_TdsOlapAggregation(spec: V1_TdsOlapAggregation): V1_ValueSpecification {
+    spec.function.accept_ValueSpecificationVisitor(this);
+    return spec;
+  }
+}
+
+function V1_resolveElementPath(
+  path: string,
+  resolverFunc: (
+    path: string,
+  ) => PackageableElementImplicitReference<PackageableElement>,
+  resolver: V1_ValueSpecificationResolver,
+): string {
+  const resolvedPath = resolverFunc(path).value.path;
+  // Note: this handles any system elements + primitve types already resolved. i.e String, ModelStore
+  if (resolvedPath !== path) {
+    resolver.hasModifiedLambda = true;
+  }
+  return resolvedPath;
+}
+
+const V1_transformV1RawLambdaToV1Lambda = (
+  v1RawLambda: V1_RawLambda,
+): V1_ValueSpecification => {
+  const v1RawLambdaJson = V1_serializeRawValueSpecification(v1RawLambda);
+  return V1_deserializeValueSpecification(v1RawLambdaJson);
+};
+
+const V1_transformV1LambdaToV1RawLambda = (
+  v1Lambda: V1_ValueSpecification,
+): V1_RawLambda => {
+  const v1LambdaJson = V1_serializeValueSpecification(v1Lambda);
+  const resolvedV1RawLambda = V1_deserializeRawValueSpecification(
+    v1LambdaJson,
+  ) as V1_RawLambda;
+  return resolvedV1RawLambda;
+};
+
+/**
+ * Method resolves element paths inside V1_RawLambda
+ * To do this we take V1_RawLamba (body: object, parameters: obect) convert it to V1_Lambda and
+ * resolve where possible (where an element path is defined as part of the value specification flow.
+ * When completed we convert it back to V1_RawLambda.
+ * This method ignores any function matching therefore if any time we encounter an error resolving an element path
+ * inside the value specification, we continue on the graph unchanging the value spec.
+ * We use this new resolved lamba if we have changed any element path (noted by the hasModifiedLambda flag)
+ * The whole flow is wrapper around a try/catch and we use the orginal lambda if any errors arise from this flow.
+ */
+
```

Comment:
Thanks for this! Though I think when you make comment like this, don't leave an empty line like this as IDE will fail to make it the comment of the function. It would be nice when you can hover on the function and see its doc.